api-server/api/routes/session.py

```diff
@@ -17,3 +30,98 @@ async def make_session():
     except Exception as e:
         logger.error("Unable to create session")
         return HTTPException(status.HTTP_500_INTERNAL_SERVER_ERROR, str(e))
+
+
+class JoinRequestBody(BaseModel):
+    username: str
+
+
+async def get_lobby(session: Session) -> Lobby:
+    lobby_players: list[LobbyPlayer] = []
+    players = await session.players.all()
+    for player in players:
+        lobby_players.append(
+            LobbyPlayer(
+                username=player.username,
+                connection_state=player.connection_state,
+                is_host=session.host == player.id,
+            )
+        )
+    return Lobby(players=lobby_players)
+
+
+@router.post(
+    "/join/{session_id}", status_code=status.HTTP_202_ACCEPTED, response_model=Lobby
+)
+async def join(session_id: str, join_body: JoinRequestBody):
+    # todo(ramko9999): select for update may prevent insertions of rows which reference session_id via Fkey. Consider passing in no_key
+    # or using an other table just for locking lobby related activities such as joining, leaving or scoring the game.
+    session = await Session.filter(id=session_id).select_for_update().first()
+    if not session:
+        raise HTTPException(
+            status.HTTP_404_NOT_FOUND, f"Session {session_id} not found"
+        )
+    player_id = session_id + "-" + join_body.username
+    async with in_transaction():
```

Comment:
is `in_transaction` like tortoise db way of executing queries in transaction?

Reply:
yes